### DISCUSSION TITLE
Not able to start a task because "VT list is empty" (but feeds are updated)

### DISCUSSION THREAD

**Discussion by sgaravat** [immauss/openvas#220] [Q&A]
Dear all

With some problems, I was eventually able to have a 22.4.28 container running and I was able to recover a database backup on this installation.

But I am not able to start tasks. This is [*] what I see in the container log
If I go to Administration --> Feed Status I see 4 feeds with status "Current" and version "20230828Txxx"
Any hints ?

Thanks, Massimo

[*]
```
event task:MESSAGE:2023-08-29 07h07.47 UTC:4106: Status of task 90.147.77.117 FIP INFN CloudProd (f9d44f37-9b34-482e-b925-fc1a3b87f1b6) has changed to Requested
event task:MESSAGE:2023-08-29 07h07.47 UTC:4106: Task 90.147.77.117 FIP INFN CloudProd (f9d44f37-9b34-482e-b925-fc1a3b87f1b6) has been requested to start by admin
md manage:WARNING:2023-08-29 07h07.47 UTC:4112: OSP start_scan 9a4bc088-33de-48ed-adb1-a82d77553c37: Exiting because VT list is empty (e.g. feed not synced yet)
event task:MESSAGE:2023-08-29 07h07.49 UTC:4112: Status of task 90.147.77.117 FIP INFN CloudProd (f9d44f37-9b34-482e-b925-fc1a3b87f1b6) has changed to Done
event task:MESSAGE:2023-08-29 07h07.50 UTC:4112: Status of task 90.147.77.117 FIP INFN CloudProd (f9d44f37-9b34-482e-b925-fc1a3b87f1b6) has changed to Interrupted

```

**Comment by immauss** [comment]
a couple of questions to work this out:.
- What is the source of the backup you restored from
- How did you accomplish the backup? 
- If you create a new task, with new targets etc ... does that run? 

Thanks,
Scott

**Comment by sgaravat** [comment]
The backup was done on another 2.4.28 installation using:
`/usr/bin/docker exec openvas su -c "/usr/lib/postgresql/13/bin/pg_dumpall" postgres > /BACKUP/db-backup-`date +%Y%m%d%H%M`.sql
`
The restore was done using:


docker create Openvas-22.4.28
docker run -it -e RESTORE=true -v /home/sgaravat/db-backup-202308281510.sql:/usr/lib/db-backup.sql --rm -v Openvas-22.4.28:/data immauss/openvas:22.4.28


Then the container was started using a docker compose file:
```
version: "3"
services:
  openvas:
    ports:
      - "9390:9390"
      - "443:9392"
    environment:
      - "PASSWORD=xyz"
      - "USERNAME=admin"
      - "RELAYHOST=192.84.143.16"
      - "SMTPPORT=25"
      - "REDISDBS=512" # number of Redis DBs to use
      - "QUIET=false"  # dump feed sync noise to /dev/null
      - "NEWDB=false"  # only use this for creating a blank DB 
      - "SKIPSYNC=false" # Skips the feed sync on startup.
      - "RESTORE=false"  # This probably not be used from compose... see docs.
      - "DEBUG=false"  # This will cause the container to stop and not actually start gvmd
      - "HTTPS=true"  # wether to use HTTPS or not
    volumes:
      Openvas-22.4.28:
       external: true
```

In this container I can see all the targets and tasks. The weird thing is that "Last report" is empty for all tasks

 I tried to create a new task, but I see the same problem: the task doesn't start and in the log it reports "\"Exiting because VT list is empty"

**Comment by immauss** [comment]
Is the "other installation of 22.4.28" still available, or has it been deleted? 

-Scott

**Comment by sgaravat** [comment]
The  "other installation of 22.4.28" is still available, but I personally don't have access to it. It is managed by a colleague

**Comment by immauss** [comment]
Can you restart the container and then attach the output of the following commands here.

```
docker logs <container name>
```
Thanks,
Scott

**Comment by sgaravat** [comment]
I restarted the container but it doesn't start anymore (I tried twice)
Please note that I am using the "beta" tag.

The docker log is attached

Thanks for your help !

Cheers, MAssimo

[
[openvas.log](https://github.com/immauss/openvas/files/12482441/openvas.log)
](url)

**Comment by immauss** [comment]
Try the latest tag.
beta is, especially right now, in constant flux and may not actually work. 

Start from scratch, with a new volume, or even no volume and see if you still get the same errors. 

-Thanks,
Scott

**Comment by sgaravat** [comment]
I restarted from scratch: new volume, latest tag, I imported an existing database. The problem with existing tasks (the ones restored from the DB) is still there. The problem is there also for new tasks created cloning old tasks. It works instead with tasks created from scratch

At any rate I eventually decided to create a new database from scratch where I recreated all targets and scans and the relevant configurations and this deployment is working.

Thanks again for your great support

**Reply by immauss** [reply]
Thanks for the follow up. Sounds like I need to validate the backup process and make sure there is nothing missing. 

Thanks,
-Scott
